src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + newLocalFileName))).isTrue();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldRenameFolder() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + currentLocalFolderName))).isTrue();
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + newLocalFolderName))).isFalse();
+        FileMove fileMove = new FileMove();
+        fileMove.setCurrentFilePath(currentLocalFolderName);
+        fileMove.setNewFilename(newLocalFolderName);
+        request.postWithoutLocation(testRepoBaseUrl + exercise.getId() + "/rename-file", fileMove, HttpStatus.OK, null);
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + currentLocalFolderName))).isFalse();
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + newLocalFolderName))).isTrue();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldDeleteFile() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + currentLocalFileName))).isTrue();
+        params.add("file", currentLocalFileName);
+        request.delete(testRepoBaseUrl + exercise.getId() + "/file", HttpStatus.OK, params);
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/" + currentLocalFileName))).isFalse();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldCommitChanges() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        var receivedStatusBeforeCommit = request.get(testRepoBaseUrl + exercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
+        assertThat(receivedStatusBeforeCommit.repositoryStatus.toString()).isEqualTo("UNCOMMITTED_CHANGES");
+        request.postWithoutLocation(testRepoBaseUrl + exercise.getId() + "/commit", null, HttpStatus.OK, null);
+        var receivedStatusAfterCommit = request.get(testRepoBaseUrl + exercise.getId(), HttpStatus.OK, RepositoryStatusDTO.class);
+        assertThat(receivedStatusAfterCommit.repositoryStatus.toString()).isEqualTo("CLEAN");
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldPullChanges() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        request.get(testRepoBaseUrl + exercise.getId() + "/pull", HttpStatus.OK, Void.class);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldResetToLastCommit() throws Exception {
+        programmingExerciseRepository.save(exercise);
```

Comment:
1) first create some "local" changes without committing them (in the best case, use the online code editor to add and modify files). If this is not possible, then directly manipulate the local git
2) Then create a merge conflict by changing the same files in the originGit
TODO: check if the Artemis server detects the conflict
3) Then invoke reset
4) Then check, that all local changes are removed and there is no conflict any more

Reply:
i adapted the method